core/src/test/java/org/apache/iceberg/hadoop/TestHadoopTables.java

```diff
@@ -92,27 +83,23 @@ public void testDropTableWithPurge() throws IOException {
         .isInstanceOf(NoSuchTableException.class)
         .hasMessageStartingWith("Table does not exist");
 
-    Assert.assertEquals(0, dataDir.listFiles().length);
-    Assert.assertFalse(tableDir.exists());
-
-    Assert.assertFalse(TABLES.dropTable(tableDir.toURI().toString()));
+    Assertions.assertThat(Files.list(dataDir.toPath()).count()).isEqualTo(0);
+    Assertions.assertThat(tableDir).doesNotExist();
+    Assertions.assertThat(TABLES.dropTable(tableDir.toURI().toString())).isFalse();
   }
 
   @Test
   public void testDropTableWithoutPurge() throws IOException {
-    File dataDir = temp.newFolder();
-
     createDummyTable(tableDir, dataDir);
 
     TABLES.dropTable(tableDir.toURI().toString(), false);
     Assertions.assertThatThrownBy(() -> TABLES.load(tableDir.toURI().toString()))
         .isInstanceOf(NoSuchTableException.class)
         .hasMessageStartingWith("Table does not exist");
 
-    Assert.assertEquals(1, dataDir.listFiles().length);
-    Assert.assertFalse(tableDir.exists());
-
-    Assert.assertFalse(TABLES.dropTable(tableDir.toURI().toString()));
+    Assertions.assertThat(Files.list(dataDir.toPath()).count()).isEqualTo(1);
```

Comment:
same here

Reply:
I've modified `Files.list(dataDir.toPath()).count()` to